packages/core/src/command/hl7v2-subscriptions/hl7v2-roster-generator.ts

```diff
@@ -4,223 +4,174 @@ import axios from "axios";
       log,
-    }
-  );
-  log(`Found ${subscribers.length} total subscribers`);
-
-  if (subscribers.length === 0) {
-    const msg = `No subscribers found, skipping roster generation`;
-    log(msg);
-    capture.message(msg, {
-      extra: config,
-      level: "info",
+      errorConfig: {
+        errorMessage: "Error uploading preprocessed CSV",
+        context: "Hl7v2RosterGenerator",
+        captureParams: config,
+        shouldCapture: true,
+      },
     });
-    return;
-  }
 
-  // TODO 2791: Make sure the IDs is scrambled in the underlying get-subscribers function
-  const convertedSubscribers = convertSubscribersToHieFormat(subscribers, hieConfig.schema);
-  const rosterCsv = generateCsv(convertedSubscribers);
-  log("Created CSV");
-
-  const fileName = buildDocumentNameForHl7v2Roster(hieConfig.name, subscriptions);
-  const s3Client = getS3UtilsInstance();
-
-  await storeInS3WithRetries({
-    s3Utils: s3Client,
-    payload: rosterCsv,
-    bucketName,
-    fileName,
-    contentType: CSV_MIME_TYPE,
-    log,
-    errorConfig: {
-      errorMessage: "Error uploading preprocessed CSV",
-      context: "Hl7v2RosterGenerator",
-      captureParams: config,
-      shouldCapture: true,
-    },
-  });
-
-  // TODO 2791: Uncomment when we update the SFTP configs to be fetched from the AWS secrets
-  // try {
-  //   await executeWithNetworkRetries(async () => sendViaSftp(hieConfig.sftpConfig, rosterCsv, log), {
-  //     maxAttempts: NUMBER_OF_ATTEMPTS,
-  //     initialDelay: BASE_DELAY.asMilliseconds(),
-  //     log,
-  //   });
-  // } catch (err) {
-  //   const msg = `Failed to SFTP upload HL7v2 roster`;
-  //   capture.error(msg, {
-  //     extra: {
-  //       hieConfig,
-  //       states,
-  //       subscriptions,
-  //       err,
-  //     },
-  //   });
-  // }
-  log("Done");
-  return;
-}
+    log("Done");
+    return rosterCsv;
+  }
 
-async function getAllSubscribers(
-  apiUrl: string,
-  states: string[],
-  subscriptions: Hl7v2Subscription[],
-  log: typeof console.log
-): Promise<Hl7v2Subscriber[]> {
-  const allSubscribers: Hl7v2Subscriber[] = [];
-  let currentUrl = `${apiUrl}/${HL7V2_SUBSCRIBERS_ENDPOINT}`;
-  let baseParams: Hl7v2SubscriberParams | undefined = {
-    states: states.join(","),
-    subscriptions,
-    count: NUMBER_OF_PATIENTS_PER_PAGE,
-  };
-
-  try {
-    while (currentUrl) {
-      const response = await axios.get(currentUrl, {
-        params: baseParams,
-      });
-      baseParams = undefined;
+  private async getAllSubscribers(
+    states: string[],
+    subscriptions: Hl7v2Subscription[],
+    log: typeof console.log
+  ): Promise<Hl7v2Subscriber[]> {
+    const allSubscribers: Hl7v2Subscriber[] = [];
+    let currentUrl = `${this.apiUrl}/${HL7V2_SUBSCRIBERS_ENDPOINT}`;
+    let baseParams: Hl7v2SubscriberParams | undefined = {
+      states: states.join(","),
+      subscriptions,
+      count: NUMBER_OF_PATIENTS_PER_PAGE,
+    };
+
+    try {
```

Comment:
Drop the try catch because it'll get caught and handled by the Sentry.AWS.wrapHandler() higher order fn.